### PR TITLE
feat(dedicated-cloud): adding location in hosts and datastores pages

### PR DESCRIPTION
**Pull request by carsso**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-14703
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Following some incidents, we want to allow customers to see in which datacenter / room / rack their resources (hosts/datastores) are.

Capture d’écran 2024-07-06 à 18 23 47
Capture d’écran 2024-07-06 à 18 23 56

## Related

MANAGER-14703

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js [hunk 1]
```diff
@@ -12,4 +12,13 @@ export default class {
       },
     );
   }
+
+  getHostLocation(serviceName, datacenterId, hostId) {
+    return this.OvhHttp.get(
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js:17]: use $http instead

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js [hunk 2]
```diff
@@ -21,4 +21,21 @@ export default class {
       },
     );
   }
+
+  getDatastoreLocation(serviceName, datacenterId, filerId) {
+    if (!datacenterId) {
+      return this.OvhHttp.get(
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js:27]: use $http instead

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js [hunk 3]
```diff
@@ -21,4 +21,21 @@ export default class {
...
+        `/dedicatedCloud/${serviceName}/filer/${filerId}/location`,
+        {
+          rootPath: 'apiv6',
+        },
+      );
+    }
+    return this.OvhHttp.get(
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js:34]: use $http instead

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js [hunk 4]
```diff
@@ -111,6 +111,20 @@ export default class {
       : this.fetchLegacyConsumption(datastores);
   }
 
+  fetchLocations(datastores) {
+    return this.$q.all(
+      datastores.map((datastore) => {
+        return this.ovhManagerPccDatacenterDatastoreService
+          .getDatastoreLocation(this.productId, datastore.dc, datastore.id)
+          .then((response) => {
+            set(datastore, 'location', response);
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js:120]: we prefer not to use lodash any more

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import get from 'lodash/get';
+import set from 'lodash/set';
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:2]: we prefer not to use lodash any more

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js [hunk 6]
```diff
@@ -105,6 +106,20 @@ export default class {
       : this.fetchLegacyHostConsumption(hosts);
   }
 
+  fetchLocations(hosts) {
+    return this.$q.all(
+      hosts.map((host) => {
+        return this.dedicatedCloudDataCenterHostService
+          .getHostLocation(this.productId, this.datacenterId, host.hostId)
+          .then((response) => {
+            set(host, 'location', response);
```
Comment (antonymarion) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:115]: we prefer not to use lodash any more

**Review by antonymarion — COMMENTED** [review]
lodash => native
ovhttp => http

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js (hunk 5 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:2]: Didn't know that, was keeping the same coding style as other files, removing it.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js (hunk 4 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js:120]: Didn't know that, was keeping the same coding style as other files, removing it.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js (hunk 3 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js:34]: Didn't know that, was keeping the same coding style as other files, switching to $http

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js (hunk 2 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js:27]: Didn't know that, was keeping the same coding style as other files, switching to $http

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js (hunk 1 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js:17]: Didn't know that, was keeping the same coding style as other files, switching to $http

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js (hunk 6 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:115]: Didn't know that, was keeping the same coding style as other files, removing it.

**Comment by carsso** [comment]
Thanks @antonymarion for the review 🙏🏻  I fixed what you reported.

I tried coding the same way I saw in other files on the same app, didn't know some things were not allowed anymore, sorry for that.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js (hunk 3 above)
Comment (antonymarion) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js]: No worries, we try to remove or code with no legacy style (all native).

Take care, no need to use rootPath with $http now I believe

Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js]: Yup, I just tried, works without rootPath, removing it now

**Review by antonymarion — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js [hunk 7]
```diff
@@ -111,6 +111,19 @@ export default class {
...
   }
 
+  fetchLocations(datastores) {
+    return this.$q.all(
+      datastores.map((datastore) => {
+        return this.ovhManagerPccDatacenterDatastoreService
+          .getDatastoreLocation(this.productId, datastore.dc, datastore.id)
+          .then((response) => {
+            return { ...datastore, location: response.data };
+          })
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js:121]:
```suggestion
          .then((location) => ( { ...datastore, location })
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html [hunk 8]
```diff
@@ -48,6 +48,21 @@ <h2 data-translate="dedicatedCloud_tab_datastores_title"></h2>
             data-ng-bind="'dedicatedCloud_tab_datastores_status_' + $row.status | translate"
         ></span>
     </oui-datagrid-column>
+    <oui-datagrid-column
+        data-title=":: 'dedicatedCloud_tab_datastores_location_datacenter' | translate"
+    >
+        <span data-ng-bind="$row.location.datacenter || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:54]: null values are not displayed if I'm not wrong

```suggestion
        <span data-ng-bind="$row.location.datacenter"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html [hunk 9]
```diff
@@ -48,6 +48,21 @@ <h2 data-translate="dedicatedCloud_tab_datastores_title"></h2>
...
+    </oui-datagrid-column>
+    <oui-datagrid-column
+        data-title=":: 'dedicatedCloud_tab_datastores_location_room' | translate"
+    >
+        <span data-ng-bind="$row.location.room || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:59]:
```suggestion
        <span data-ng-bind="$row.location.room"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html [hunk 10]
```diff
@@ -48,6 +48,21 @@ <h2 data-translate="dedicatedCloud_tab_datastores_title"></h2>
...
+    </oui-datagrid-column>
+    <oui-datagrid-column
+        data-title=":: 'dedicatedCloud_tab_datastores_location_rack' | translate"
+    >
+        <span data-ng-bind="$row.location.rack || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:64]:
```suggestion
        <span data-ng-bind="$row.location.rack"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js [hunk 11]
```diff
@@ -105,6 +105,19 @@ export default class {
...
   }
 
+  fetchLocations(hosts) {
+    return this.$q.all(
+      hosts.map((host) => {
+        return this.dedicatedCloudDataCenterHostService
+          .getHostLocation(this.productId, this.datacenterId, host.hostId)
+          .then((response) => {
+            return { ...host, location: response.data };
+          })
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:115]:
```suggestion
          .then((location) => ({  ...host, location }))
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html [hunk 12]
```diff
@@ -45,6 +45,21 @@ <h2 data-translate="dedicatedCloud_tab_host"></h2>
                     data-translate="{{ 'dedicatedCloud_tab_host_status_' + $row.status }}"
                 ></span>
             </oui-datagrid-column>
+            <oui-datagrid-column
+                data-title=":: 'dedicatedCloud_tab_host_location_datacenter' | translate"
+            >
+                <span data-ng-bind="$row.location.datacenter || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html:51]:
```suggestion
                <span data-ng-bind="$row.location.datacenter"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html [hunk 13]
```diff
@@ -45,6 +45,21 @@ <h2 data-translate="dedicatedCloud_tab_host"></h2>
...
+            </oui-datagrid-column>
+            <oui-datagrid-column
+                data-title=":: 'dedicatedCloud_tab_host_location_room' | translate"
+            >
+                <span data-ng-bind="$row.location.room || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html:56]:
```suggestion
                <span data-ng-bind="$row.location.room"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html [hunk 14]
```diff
@@ -45,6 +45,21 @@ <h2 data-translate="dedicatedCloud_tab_host"></h2>
...
+            </oui-datagrid-column>
+            <oui-datagrid-column
+                data-title=":: 'dedicatedCloud_tab_host_location_rack' | translate"
+            >
+                <span data-ng-bind="$row.location.rack || ''"></span>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html:61]:
```suggestion
                <span data-ng-bind="$row.location.rack"></span>
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js [hunk 15]
```diff
@@ -12,4 +13,10 @@ export default class {
       },
     );
   }
+
+  getHostLocation(serviceName, datacenterId, hostId) {
+    return this.$http.get(
+      `/dedicatedCloud/${serviceName}/datacenter/${datacenterId}/host/${hostId}/location`,
+    );
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.service.js:20]:
```suggestion
    return this.$http.get(
      `/dedicatedCloud/${serviceName}/datacenter/${datacenterId}/host/${hostId}/location`,
    ).then(({data}) => data);
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js [hunk 16]
```diff
@@ -21,4 +22,15 @@ export default class {
...
+
+  getDatastoreLocation(serviceName, datacenterId, filerId) {
+    if (!datacenterId) {
+      return this.$http.get(
+        `/dedicatedCloud/${serviceName}/filer/${filerId}/location`,
+      );
+    }
+    return this.$http.get(
+      `/dedicatedCloud/${serviceName}/datacenter/${datacenterId}/filer/${filerId}/location`,
+    );
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js:34]:
```suggestion
      return this.$http.get(
        `/dedicatedCloud/${serviceName}/filer/${filerId}/location`,
      ).then(({data}) => data);
    }
    return this.$http.get(
      `/dedicatedCloud/${serviceName}/datacenter/${datacenterId}/filer/${filerId}/location`,
    ).then(({data}) => data);
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js (hunk 7 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js:121]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html (hunk 8 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:54]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html (hunk 9 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:59]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html (hunk 10 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.html:64]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js (hunk 11 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js:115]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html (hunk 12 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html:51]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html (hunk 13 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html:56]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html (hunk 14 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/host/dedicatedCloud-datacenter-host.html]: done, thanks

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js (hunk 16 above)
Comment (carsso) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.service.js]: done, thanks

**Review by anooparveti — DISMISSED** [review]

**Review by MaximeBajeux — APPROVED** [review]

**Pull request merged**